src/BccCode.Linq/ApiClient/ResultList.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System.Collections;
+using System.ComponentModel.DataAnnotations;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace BccCode.ApiClient;
+
+[DataContract(Name = "ResultList")]
+public abstract class ResultList
+{
+    protected ResultList(Metadata? meta = null)
+    {
+    }
+
+    /// <summary>
+    /// Gets or Sets Meta
+    /// </summary>
+    [DataMember(Name = "meta", EmitDefaultValue = false)]
+    public Metadata? Meta { get; set; }
+
+    [JsonIgnore]
+    public IEnumerable<object>? Data { get; set; }
+
+    public abstract void AddData(IEnumerable? data);
+
+    [Obsolete($"Use property {nameof(Data)} instead")]
+    public virtual IEnumerable<object> GetData() => Data ?? new List<object>();
+
+    [Obsolete($"Use property {nameof(Data)}.Count() instead")]
+    public virtual long GetCount() => Data?.LongCount() ?? 0;
+}
+
+
+public class ResultList<T> : ResultList, IResultList<T>, IEquatable<ResultList<T>>, IValidatableObject
+{
+    public ResultList()
+    {
+    }
+
+    public ResultList(ResultList<T> resultList)
+    {
+        Data = resultList.Data;
+        Meta = resultList.Meta;
+    }
+
+    public ResultList(IResultList<T> resultList)
+    {
+        if (resultList.Data is List<T> dataList)
+        {
+            Data = dataList;
+        }
+        else
+        {
+            Data = new List<T>(resultList.Data);
+        }
+
+        if (resultList.Meta is Metadata metadata)
+        {
+            Meta = metadata;
+        }
+        else
+        {
+            Meta = new Metadata(resultList.Meta);
+        }
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ResultList" /> class.
+    /// </summary>
+    /// <param name="data">data.</param>
+    /// <param name="meta">meta.</param>
+    public ResultList(List<T>? data = default, Metadata? meta = default)
+    {
+        this.Data = data;
+        this.Meta = meta;
+    }
+
+    /// <summary>
+    /// Gets or Sets Data
+    /// </summary>
+    [DataMember(Name = "data", EmitDefaultValue = false)]
+    public new List<T>? Data { get; set; }
+
+    public override void AddData(IEnumerable? data)
+    {
+        Data ??= new List<T>();
+        if (data != null)
+        {
+            Data.AddRange(data.Cast<T>());
+        }
+    }
+
+    [Obsolete("Use property Data instead")]
+    public override IEnumerable<object> GetData()
```

Comment:
No need to make this obsolete.

Reply:
My idea was to use `Data` instead of `GetData()`. As commented above, in general, I do not know if we really need that `ResultList` base class. `ResultList<T>` would be sufficient in my opinion.

Reply:
The reason for the base class was to allow merging multiple results of an unknown type (without using reflection).

Reply:
Since we dropped now `ResultList`, I think we should keep now the `Obsolete` attributes.